components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/client/authentication/OAuthClientAuthenticator.java

```diff
@@ -65,4 +65,11 @@ boolean canAuthenticate(HttpServletRequest request, Map<String, List> bodyParams
      */
     String getClientId(HttpServletRequest request, Map<String, List> bodyParams, OAuthClientAuthnContext
             oAuthClientAuthnContext) throws OAuthClientAuthnException;
+
+    /**
+     * Retrieve the referred name of the authenticator.
+     *
+     * @return      Referred name of the authenticator.
+     */
+    String getReferredName();
```

Comment:
Can we add it as default method, so it won't affect existing implementations?

Reply:
Fixed in 684ce335e1241bfce1d6fd56aae4ca08abba580c